### PR TITLE
implement a log of events queriable from the metrics endpoint

### LINKED ISSUE #304: Query marble launch log from http
Client doesn't want to use kubernetes but wants to query the "state" somewhere, meaning the last time a marble was launched. 

We'd implement this using the existing metrics service  just returning an /events endpoint with a list of the relevant log items.

Is this something acceptable upstream?

### PR DESCRIPTION
**Pull request by aep**
fixes #304

### CODE REVIEW

**Review by m1ghtym0 — APPROVED** [review]
lgtm:-)

File: coordinator/events/eventlog.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Edgeless Systems GmbH.
+//
```
Comment (daniel-weisse) [inline coordinator/events/eventlog.go:2]: Please run `gofmt -w coordinator/events/eventlog.go` to format this file.

Also all top-level, exported types and functions should have doc comments.

File: coordinator/core/core.go [hunk 2]
```diff
@@ -112,7 +114,7 @@ func (c *Core) advanceState(newState state, tx store.Transaction) error {
 }
 
 // NewCore creates and initializes a new Core object.
-func NewCore(dnsNames []string, qv quote.Validator, qi quote.Issuer, sealer seal.Sealer, recovery recovery.Recovery, zapLogger *zap.Logger, promFactory *promauto.Factory) (*Core, error) {
+func NewCore(dnsNames []string, qv quote.Validator, qi quote.Issuer, sealer seal.Sealer, recovery recovery.Recovery, zapLogger *zap.Logger, promFactory *promauto.Factory, eventlog *events.Log) (*Core, error) {
```
Comment (daniel-weisse) [inline coordinator/core/core.go:117]: Unit tests that call `NewCore()` need to be update.

File: coordinator/events/eventlog.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	})
+}
+
+func (ev *Log) Handler() http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		json.NewEncoder(w).Encode(ev.events)
+	})
+}
```
Comment (daniel-weisse) [inline coordinator/events/eventlog.go:47]:
```suggestion
func (l *Log) Activation(marbleType string, uuid string, quote []byte) {
	l.events = append(l.events, Event{
		Timestamp:	time.Now(),
		Activation: &ActivationEvent{MarbleType: marbleType, UUID: uuid, Quote: quote},
	})
}

func (l *Log) Handler() http.HandlerFunc {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		w.WriteHeader(http.StatusOK)
		json.NewEncoder(w).Encode(l.events)
	})
}
```
Just a small style nitpick

**Review by daniel-weisse — CHANGES_REQUESTED** [review]

File: coordinator/events/eventlog.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+// Activation adds an activation event to the log.
+func (l *Log) Activation(marbleType string, uuid string, quote []byte) {
+	l.events = append(l.events, Event{
+		Timestamp:  time.Now(),
+		Activation: &ActivationEvent{MarbleType: marbleType, UUID: uuid, Quote: quote},
+	})
+}
+
+// The http handler retutns the log as JSON array
+func (l *Log) Handler() http.HandlerFunc {
```
Comment (daniel-weisse) [inline coordinator/events/eventlog.go:48]:
```suggestion
// Handler returns a http.HandlerFunc which writes the log as JSON array.
func (l *Log) Handler() http.HandlerFunc {
```

**Review by daniel-weisse — APPROVED** [review]

**Pull request merged**